app/code/community/Bolt/Boltpay/Block/Checkout/Boltpay.php

```diff
@@ -134,6 +103,52 @@ function() {
         return $boltHelper->transmit('orders', $orderRequest);
     }
 
+
+    /**
+     * Get Promise to create an order token
+     *
+     * @param string  $checkoutType  'multi-page' | 'one-page' | 'admin' | 'firecheckout'
+     *
+     * @return string javascript Promise string used for initializing BoltCheckout
+     */
+    protected function getBoltOrderTokenPromise($checkoutType) {
+
+        if ( $checkoutType === self::CHECKOUT_TYPE_FIRECHECKOUT ) {
+            $checkoutTokenUrl = Mage::helper('boltpay/url')->getMagentoUrl('boltpay/order/firecheckoutcreate');
+            $parameters = 'checkout.getFormData ? checkout.getFormData() : Form.serialize(checkout.form, true)';
+        } else if ( $checkoutType === self::CHECKOUT_TYPE_ADMIN ) {
+            $checkoutTokenUrl = Mage::helper('boltpay/url')->getMagentoUrl("adminhtml/sales_order_create/create/checkoutType/$checkoutType", array(), true);
+            $parameters = "''";
+        } else {
+            $checkoutTokenUrl = Mage::helper('boltpay/url')->getMagentoUrl("boltpay/order/create/checkoutType/$checkoutType");
+            $parameters = "''";
+        }
+
+        return <<<PROMISE
+                    new Promise( 
+                        function (resolve, reject) {
+                            new Ajax.Request('$checkoutTokenUrl', {
+                                method:'post',
+                                parameters: $parameters,
+                                onSuccess: function(response) {
+                                    if(response.responseJSON.error) {                                                        
+                                        reject(response.responseJSON.error_messages);
+                                        
+                                        // BoltCheckout is currently not doing anything reasonable to alert the user of a problem, so we will do something as a backup
+                                        alert(response.responseJSON.error_messages);
+                                        location.reload();
```

Comment:
As implemented, since the event that we are relying upon is form submission, there should be no problem here in the average use case as the retry is triggered by user interaction with a form.  Alert and reload was an old approach to the Bolt modal not supplying sufficient mechanisms for displaying a message to the user and closing the Bolt modal.

I will implement these and do this in the normal, elegant way.  When the Bolt modal code is updated, then we can use those mechanisms.

Reply:
Since Promise is no longer associated with modal open, `reload` is being removed.